modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -217,84 +164,103 @@ def get_files_info(ip, _rport, shares, info)
         next
       end
 
       if !datastore['ShowFiles']
-        print_status("Spidering #{x}.")
+        print_status("Spidering #{share_name}.")
       end
+
+      begin
+        tree = simple.client.tree_connect("\\\\#{ip}\\#{share_name}")
+      rescue RubySMB::Error::UnexpectedStatusCode => e
+        print_error "Error when trying to connect to share #{share_name} - #{e.status_code.name}"
+        print_status "Spider #{share_name} complete."
+        next
+      end
+
       subdirs = ['']
-      if (x.strip == 'C$') && datastore['SpiderProfiles']
-        subdirs = profile_options(ip, x)
+      if (share_name == 'C$') && datastore['SpiderProfiles']
+        subdirs = profile_options(tree, share[:type])
       end
       until subdirs.empty?
-        depth = subdirs[0].count('\\')
-        if datastore['SpiderProfiles'] && (x == 'C$')
-          if depth - 2 > datastore['MaxDepth']
+        depth = subdirs.first.count('\\')
+
+        if share_name == 'C$'
+          if datastore['SpiderProfiles']
+            if (depth - 2) > datastore['MaxDepth']
+              subdirs.shift
+              next
+            end
+          else
             subdirs.shift
             next
           end
-        elsif depth > datastore['MaxDepth']
+        end
+
+        if depth > datastore['MaxDepth']
           subdirs.shift
           next
         end
-        read, write, type, files = eval_host(ip, x, subdirs[0])
+
+        read, write, _type, files = eval_tree(tree, share[:type], subdirs.first)
+
         if files && (read || write)
-          if files.length < 3
+          if files.empty?
             subdirs.shift
             next
           end
-          header = ''
-          if simple.client.default_domain && simple.client.default_name
-            header << " \\\\#{simple.client.default_domain}"
-          end
-          header << "\\#{x.sub('C$', 'C$\\')}" if simple.client.default_name
-          header << subdirs[0]
 
+          header = ''
           pretty_tbl = Rex::Text::Table.new(
             'Header' => header,
             'Indent' => 1,
             'Columns' => [ 'Type', 'Name', 'Created', 'Accessed', 'Written', 'Changed', 'Size' ]
           )
 
-          f_types = {
-            1 => 'RO', 2 => 'HIDDEN', 4 => 'SYS', 8 => 'VOL',
-            16 => 'DIR', 32 => 'ARC', 64 => 'DEV', 128 => 'FILE'
-          }
+          if simple.client.default_domain && simple.client.default_name
+            header << " \\\\#{simple.client.default_domain}"
+          end
+          header << "\\#{share_name.sub('C$', 'C$\\')}" if simple.client.default_name
+          header << subdirs.first
 
           files.each do |file|
-            next unless file[0] && (file[0] != '.') && (file[0] != '..')
+            fname = file.file_name.encode('UTF-8')
+            tcr = file.create_time.to_datetime
+            tac = file.last_access.to_datetime
+            twr = file.last_write.to_datetime
+            tch = file.last_change.to_datetime
```

Comment:
is `to_datetime` the same as `to_unix_time` 👀

Reply:
They are different methods 👍 `to_unix_time` gives us very, very different output when compared to the correct output of `to_datetime`. `to_datetime` is the RubySMB-specific function that handles it correctly.

Reply:
Looks good to me

SMB returns this field:
https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-cifs/94ef49b8-648c-47f1-9c64-def4ad0a63a0?redirectedfrom=MSDN

Which is handled correctly here:

https://github.com/rapid7/ruby_smb/blob/fd6d70e3e6697be0c3815d02e216466bbd7d6bda/lib/ruby_smb/field/utime.rb#L39-L45

The new date format is in a non-ambiguous date format too, which is good :+1: